PhysicsTools/PatAlgos/plugins/PATMETProducer.cc

```diff
@@ -47,6 +47,10 @@ PATMETProducer::PATMETProducer(const edm::ParameterSet & iConfig):
   if(calculateMETSignificance_)
     {
       metSigAlgo_ = new metsig::METSignificance(iConfig);
+      rhoToken_ = mayConsume<double>(iConfig.getParameter<edm::InputTag>("srcRho"));
```

Comment:
from the implementation it looks like srcRho is always consumed if calculateMETSignificance_ is set.
If that's the case, use "consumes" instead of mayConsume.

The same seems to be true for jetToken_ and pfCandToken_ and even leptons as well.

The code will throw if metSource returns an empty view.
So, better let the framework know that these "mayConsume" are actually "consume" so that it can schedule the module execution better.